minibot/new_minibot.py

```diff
@@ -115,21 +123,27 @@ def process_string(value):
     return program
 
 
-def spawn_script_process(scriptname):
+def spawn_script_process(scriptname, tcpInstance):
```

Comment:
Hmm so we had to pass the tcpInstance in the end?  There was no way around it?

Reply:
It seems like a function run on another process cannot change global variables, so this is probably the only way?

Reply:
(I ran a little experiment myself)

Reply:
Ohhh interesting, we may have to use shared memory to handle it.  No worries, Swathi, Brandon and i can look into when we're cleaning up the backend.  You have done most of the hard work so these small things will be easy to change :).